client/components/site-selector/index.jsx

```diff
@@ -311,13 +311,27 @@ export class SiteSelector extends Component {
 		return sites;
 	}
 
+	mapAllSitesPath = ( path ) => {
+		if ( path.includes( '/posts/my' ) ) {
+			return path.replace( '/posts/my', '/posts' );
+		}
+
+		// We used to show multisite plugins management on /plugins route, but that has changed
+		// and now we show the plugins marketplace on it, which is a single site view.
```

Comment:
Yes, I find the fact that the new functionality overrode the previous behavior a bit confusing too, both for `/plugins` and `/plugins/:siteSlug`. (we had Calypso view for this too, and now it seems to be forcing wp-admin)

Looking at that page a bit more, maybe "site-less" would be a better description since it shows pretty much the same view regardless of the site, so `/plugins/:siteSlug` wouldn't be a good fit either. 😄